js/app.js

```diff
@@ -46,26 +57,34 @@ function prepareStageElement(parentClassName, className, elementType) {
 
 function text(parentClassName, className, elementType, text, delay = DEFAULT_DELAY) {
     return () => {
-        element = prepareStageElement(parentClassName, className, elementType)
+        let element = prepareStageElement(parentClassName, className, elementType)
         return new Promise((resolve, _) => {
             renderText(element, resolve, text, delay)
         })
     }
 }
 
-function custom(parentClassName, className, elementType, f) {
+function createCursor(refElement) {
+    const cursor = refElement.insertBefore(document.createElement("span"), null);
+    cursor.id = "cursor";
+    cursor.className = "cursor";
+    cursor.style.fontSize = refElement.style.fontSize;
+    cursor.innerHTML = "|";
+}
+
+function custom(parentClassName, className, elementType, f, withCursor = true) {
     return () => {
-        return new Promise(f(prepareStageElement(parentClassName, className, elementType)))
+        return new Promise(f(prepareStageElement(parentClassName, className, elementType, withCursor)))
```

Comment:
```suggestion
        let element = prepareStageElement(parentClassName, className, elementType)
        return new Promise(f(element))
```

looks better that way

Reply:
This will simply not work with your architecture of method calls and promises.

Reply:
I get my words back, it works, but the `withCursor` argument should remain.